packages/@okta/babel-plugin-handlebars-inline-precompile/handlebars/patch-precompile.js

```diff
@@ -0,0 +1,17 @@
+// Monkey patch handlebars to remove extra whitespace from templates before they are precompiled
+module.exports = function (Handlebars) {
+  const precompile = Handlebars.precompile;
+  Handlebars.precompile = function (template) {
+    // Trim extra whitespace from template
+    template = template
+      .replace(/\s+</g, '<') // all spaces before opening tag
+      .replace(/>\s+/g, '>') // all spaces after closing tag
+      .replace(/}}\s+{{/g, '}}{{') // all spaces betwen mustaches
```

Comment:
Is this a safe operation? In admin we have many templates that rely on chaining one value to another with a space in between. Looks like the closest occurrence here (uses a slightly different format) is in [FactorsDropDownOptions.js](https://github.com/okta/okta-signin-widget/blob/d294c668f1a2fffb5078bd2f9ef72f39cb400d2b/src/views/mfa-verify/dropdown/FactorsDropDownOptions.js#L21).

If we omit this substitution, does anything break?

Reply:
This file is copied from `okta-ui` by the `courage-for-signin-widget` build. There is a separate PR for these changes over here: https://github.com/okta/okta-ui/pull/2127

This prevents extra (unintentional) whitespace in the precompiled templates. Without this, the bundle size is larger and also many tests would need to be changed to account for extra whitespace.

Reply:
Both the changes proposed in [okta/okta-ui#2127](https://github.com/okta/okta-ui/pull/2127) and here have this potential side-effect.

In `okta-ui`, I'd assume we'd need to remove this line to make sure chained template values that expect spaces are met.

Examples:
- [PendingRequestsView.js](https://github.com/okta/okta-ui/blob/master/packages/admin/appinstance/src/assignments/views/PendingRequestsView.js#L72)
- [DevicesList.js](https://github.com/okta/okta-ui/blob/master/packages/admin/devices/src/views/DevicesList.js#L53)
- [UserPicker.js](https://github.com/okta/okta-ui/blob/master/packages/admin/shared/src/views/ud/mappings/components/UserPicker.js#L30)
- [SearchUtil](https://github.com/okta/okta-ui/blob/master/packages/admin/syslog/src/util/SearchUtil.js#L95)

How much extra of a bundle size does the removal of this space save? Refactoring our template logic to include spaces at the `*.properties` or passed param level (i.e `{ space: '  '}`) doesn't seem like the best option.

cc: @rajaguruduraisamy-okta

Reply:
👋🏻  Hi everyone, coming to you from the future as someone who has been tasked with updating SearchUtil.js to use precompiled templates and I just want to say that @jmelberg-okta was absolutely right this change is problematic there and the only way to proceed is to replace handlebars with es6 template literals.